pkg/controller/cainjector/setup.go

```diff
@@ -67,37 +77,85 @@ var (
 
 // registerAllInjectors registers all injectors and based on the
 // graduation state of the injector decides how to log no kind/resource match errors
-func registerAllInjectors(mgr ctrl.Manager, sources ...caDataSource) error {
+func registerAllInjectors(ctx context.Context, groupName string, mgr ctrl.Manager, sources []caDataSource, client client.Client, ca cache.Cache) error {
+	controllers := map[string]controller.Controller{}
```

Comment:
Set up all the controllers and their caches and watches before starting them later.